### PR TITLE
Implement capture for generators

### PR DESCRIPTION
**Pull request by NickeZ**
Not sure if you want this or not. I need it for a project I'm trying to convert to meson..

### CODE REVIEW

File: mesonbuild/backend/ninjabackend.py [hunk 1]
```diff
@@ -1813,6 +1813,20 @@ def generate_genlist_for_target(self, genlist, target, outfile):
...
             args = self.replace_paths(target, args)
             cmdlist = exe_arr + self.replace_extra_args(args, genlist)
+            if generator.capture:
+                exe_data = self.serialize_executable(
+                    cmdlist[0],
+                    cmdlist[1:],
+                    self.environment.get_build_dir(),
+                    capture=outfiles[0]
+                )
+                print(self.environment.get_build_dir())
```
Comment (jpakkane) [inline mesonbuild/backend/ninjabackend.py]: Forgotten debug print?

**Comment by jpakkane** [comment]
This is nice, it should get a mention in the release notes as well. Please add a snippet file for this.

**Comment by NickeZ** [comment]
Removed the debug print, added snippet in release notes.

Please review this carefully. I basically copied code from `custom_target` that already had `capture`. It works for me at least..

**Comment by nirbheek** [comment]
CI is failing because you didn't implement it for the VS backend.

**Comment by NickeZ** [comment]
Hopefully fixed it. tried it on vs2015.

**Comment by NickeZ** [comment]
Rebased, should be ready for code review/merge

**Pull request merged**